cmd/docker/main.go

```diff
@@ -130,16 +136,30 @@ func handleFlags() bool {
 	}
 	extraconfig.Decode(src, &vchConfig)
 
-	if *cli.debug || vchConfig.Diagnostics.DebugLevel > 0 {
-		log.SetLevel(log.DebugLevel)
-	}
-
 	*cli.portLayerAddr = fmt.Sprintf("%s:%d", *cli.portLayerAddr, *cli.portLayerPort)
 	cli.proto = "tcp"
 
 	return true
 }
 
+func initLogging() error {
+	logcfg := viclog.NewLoggingConfig()
+	if *cli.debug || vchConfig.Diagnostics.DebugLevel > 0 {
+		logcfg.Level = log.DebugLevel
+	}
+
+	if vchConfig.Diagnostics.SysLogConfig != nil {
+		logcfg.Syslog = &syslog.SyslogConfig{
+			Network:   vchConfig.Diagnostics.SysLogConfig.Network,
+			RAddr:     vchConfig.Diagnostics.SysLogConfig.RAddr,
+			Formatter: syslog.RFC3164,
+			Priority:  srslog.LOG_INFO | srslog.LOG_DAEMON,
+		}
+	}
+
+	return viclog.Init(logcfg)
```

Comment:
We can put out log messages here at the beginning here, but once we are inside the syslog hook, we can't produce any more log messages since they end up again in the syslog loop, thus resulting in a positive feedback loop that will run away exponentially.

The syslog library code will keep retrying if the remote goes down or is down to begin with. We redial if we never have a `srslog.Writer` to begin with -- the case where the remote is down at the start. Once we have a `srslog.Writer` object, it will retry on its own, and we don't have to redial.